CONTRIBUTING.md

```diff
@@ -1,57 +1,95 @@
- * try to add a title that describe succinctly the issue
- * if you are running your own registry, please provide:
-  * registry version
-  * registry launch command used
-  * registry configuration
-  * registry logs
- * in all cases:
-  * `docker version` and `docker info`
-  * run your docker daemon in debug mode (-D), and provide docker daemon logs 
+## Reporting an issue properly
 
-## You have a patch for a known bug, or a small correction?
+By following these simple rules you will get better and faster feedback on your issue.
 
-Basic github workflow (fork, patch, make sure the tests pass, PR).
+ - [ ] search the bugtracker for an already reported issue
 
-... and some simple rules to ensure quick merge:
+### If you found an issue that describes your problem:
 
- * clearly point to the issue(s) you want to fix
- * when possible, prefer multiple (smaller) PRs addressing individual issues over a big one trying to address multiple issues at once
- * if you need to amend your PR following comments, squash instead of adding more commits
+ - [ ] please read other user comments first, and confirm this is the same issue: a given error condition might be indicative of different problems - you may also find a workaround in the comments
+ - [ ] please refrain from adding "same thing here" or "+1" comments
+ - [ ] you don't need to comment on an issue to get notified of updates: just hit the "subscribe" button
+ - [ ] comment if you have some new, technical and relevant information to add to the case
 
-## You want some shiny new feature to be added?
+### If you have not found an existing issue that describes your problem:
 
-Fork the project.
+ - [ ] create a new issue, with a succinct title that describes your issue:
+   * bad title: "It doesn't work with my docker"
+   * good title: "Private registry push fail: 400 error with E_INVALID_DIGEST"
+ - [ ] copy the output of `docker version`
+ - [ ] copy the output of `docker info`
+ - [ ] copy your Registry version
+ - [ ] copy the command line you used to launch your Registry
+ - [ ] restart your docker daemon in debug mode (add `-D` to the daemon launch arguments)
+ - [ ] reproduce your problem and get your docker daemon logs showing the error
+ - [ ] if relevant, copy your registry logs that show the error
+ - [ ] provide any relevant detail about your specific Registry configuration (eg: storage backend used)
+ - [ ] indicate if you are using an enterprise proxy, Nginx, or anything else between you and your Registry
 
-Create a new proposal in the folder `open-design/specs`, named `DEP_MY_AWESOME_PROPOSAL.md`, using `open-design/specs/TEMPLATE.md` as a starting point.
+## Contributing a patch for a known bug, or a small correction
 
-Then immediately submit this new file as a pull-request, in order to get early feedback.
+You should follow the basic GitHub workflow:
 
-Eventually, you will have to update your proposal to accommodate the feedback you received.
+ - [ ] fork
+ - [ ] commit a change
```

Comment:
I don't think that could be the cause of any issue.
@stevvooe what do you think? Should we be pressing about the use of feature branches inside the user fork?